---
title: "A1.2"
output: html_document
date: "2025-09-19"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r}
library(susieR)
library(dplyr)
library(ggplot2)
library(data.table)
library(pheatmap)

```
The data can be found in the directory (/course/advBinf/rna/data) directory on Emily


```{r}

gwas <- fread("/course/advBinf/rna/data/height_wood.ma")
head(gwas)
summary(gwas)
```


```{r}
locus <- gwas %>%
  filter(CHR == 18 & POS >= 46306107 & POS <= 47306107)
```
Q1: What does the GWAS data at the extracted locus look like?
The plot below shows the variation in p values over the different positions of the specified locus. I notice a wide area of potentially interesting SNP:s. 

```{r}
ggplot(locus, aes(x=POS, y=-log10(P))) +
  geom_point(alpha=0.4, size=1) +
  geom_label_repel(data=subset(locus, locus$SNP=="rs955748"), 
                   aes(label=SNP), size=4, max.overlaps = 20)
```
Q2: Is there a clear lead variant?, explain.
From this plot there are several potential lead variants. some of the points are reaching toward -log(p) values of almost 40 which is promising. However we need more information such as LD in order to say anything for sure.

Construct LD (linkeage disequilibrium) matrix.
```{r eval=FALSE}
write.table(locus$SNP, file="~/locus_snps.txt", 
            col.names=F, row.names=F, quote=F)
```


```{bash eval=FALSE}
plink2\
  --bfile /course/advBinf/rna/data/g1000_eur \
  --r-phased square ref-based \
  --extract ~/locus_snps.txt \
  --out ~/locus_1000G
```


```{r eval=FALSE}
locus.ld <- read.table("./locus_1000G.phased.vcor1")
head(locus.ld[,c(1:5)])
```

Q3: What does the LD structure of the locus look like?
According the heat-map and the chosen significance level there are a few s significant sites around the left of the center of the locus. similar as to what was seen in the previous plot. The correlation cofefficients in the center is larger which indicates that there are pairs of aleles there which tend to be inherited together.

```{r eval=FALSE}
my_sample_col <- data.frame(significant = as.factor(-log10(locus$P)>5))
row.names(my_sample_col) <- row.names(locus.ld) <- colnames(locus.ld)

pheatmap(locus.ld, cluster_rows=F, cluster_cols=F, show_colnames = FALSE, 
         show_rownames = FALSE,annotation_row = my_sample_col, 
         annotation_col = my_sample_col)
```

Q4: Why do we use the European samples from the 1000G project for calculating LD?
LD will vary between populations and because the locus data we are analyzing is from a study on european individuals we should use european individuals as a reference.



```{r eval=FALSE}
ld.mat = matrix(as.vector(data.matrix(locus.ld)), 
                nrow=nrow(locus.ld), ncol=nrow(locus.ld))
```

```{r eval=FALSE}
fitted_rss = susie_rss(bhat = locus$b, R = ld.mat, shat = locus$se, n = median(locus$N))
```
Q6: How many credible sets do we get from SusieR?
In the following print we can se that we have 4 credible sets.
Q7: How many variants are within the credible sets from SusieR?
set1:
1 variant. highly likley that it is causal.
set2:
12 variants, which are all correlated which means that they have similar LD. it is therefore difficult to say which one is causal
set3:
2 variants, same situation as set 2 but only fewer ones to choose from
set4:
1 variant. highly likley that it is causal.

```{r eval=FALSE}
print(fitted_rss$sets)
```


```{r eval=FALSE}
locus$variable = 1:nrow(locus)

locus_susie = merge(x=locus, y=summary(fitted_rss)$vars, 
             by="variable", all.x=TRUE)

ggplot(locus_susie, aes(x=POS, y=-log10(P))) +
  geom_point(alpha=0.5, size=1) +
  geom_point(data=subset(locus_susie, locus_susie$cs!=-1), aes(color=cs), size=4, alpha = 0.5) +
  geom_label_repel( data=subset(locus_susie, locus_susie$cs!=-1), aes(label=SNP), size=4, max.overlaps = 30, alpha = 0.7) 
```

Q8: What does the distribution of PIP values look like and what are the most likely causal variants based on it?
According to the summary() and the plot below we can see that variable 32 and 454 which likely correspond to SNP:s rs1319945 and rs9967417 respectively. i would also include credible set 2 since it is almost certain that one of these are causal, however the model can not distinguish between them. credible set 2 can also be investigated however they are more ambiguous.

```{r eval=FALSE}
summary(fitted_rss)
```

```{r eval=FALSE}
ggplot(locus_susie, aes(x=POS, y=variable_prob)) +
  geom_point(alpha=0.4, size=1.0) +
  geom_point(data=subset(locus_susie, locus_susie$cs!=-1), aes(color=cs), size=4, alpha = 0.5) +
  geom_label_repel(data=subset(locus_susie, locus_susie$cs!=-1), aes(label=SNP), size=4, max.overlaps = 20, alpha = 0.7)
```




```{r eval=FALSE}
write.table(locus[,c(1:8)], file="locus_sumstats.txt", col.names=T, row.names=F, quote=F)
```

```{bash, eval=FALSE}
/course/advBinf/rna/tools/smr  \
 --extract-snp locus_snps.txt  \
 --bfile /course/advBinf/rna/data/g1000_eur  \
 --gwas-summary locus_sumstats.txt  \
 --beqtl-summary /course/advBinf/rna/data/cis-eQTLs-full_eQTLGen \
 --thread-num 10  \
 --out locussmr   
```

Q9: What do the SMR results look like? Explain what the most important columns contain (Hint: Focus on those you have not seen before)
The results from smr run can be seen below, now i am lookin for variants with low P_smr and high P_heidi


```{r eval=FALSE}
locus.smr <- read.table("locussmr.smr",header=T)

head(locus.smr)
```


```{r}
# Read in SMR results
locus.smr <- read.table("locussmr.smr", header = TRUE)

# Define significance for P_SMR and HEIDI
locus.smr$smr_significant <- locus.smr$p_SMR < 5e-8
locus.smr$heidi_pass <- locus.smr$p_HEIDI > 0.05  # TRUE if HEIDI does not reject pleiotropy

# Combine flags for coloring
locus.smr$highlight <- locus.smr$smr_significant & locus.smr$heidi_pass

# Plot
ggplot(locus.smr, aes(x = p_HEIDI, y = -log10(p_SMR))) +
  geom_point(alpha = 0.4, size = 1.0) +  # all points faint
  geom_point(data = subset(locus.smr, highlight), aes(color = highlight),
             size = 4, alpha = 0.5) +   # highlight SMR+HEIDI good variants
  geom_label_repel(aes(label = topSNP), 
                   size = 4, max.overlaps = 20, alpha = 0.7) +
  scale_y_continuous(trans = 'log10') +  # log scale for P_SMR
  scale_color_manual(values = c("TRUE" = "red")) +  # highlighted points in red
  labs(x = "p_HEIDI", y = "p_SMR (log10 scale)", 
       title = "SMR results: p_SMR vs p_HEIDI") +
  theme_bw() +
  theme(legend.position = "none")
```
Q10: What is the most likely causal variant based on the SMR/HEIDI results?
The most likley causal variant is "rs549201", it has low p_SMR and p_HEIDI > 0.05.

```{r eval=FALSE}
ggplot(locus_susie, aes(x=POS, y=variable_prob)) +
  geom_point(alpha=0.4, size=1.0) +
  geom_point(data=subset(locus_susie, locus_susie$cs!=-1), aes(color=cs), size=4, alpha = 0.5) +
  geom_label_repel(data=subset(locus_susie, locus_susie$cs!=-1), aes(label=SNP), size=4, max.overlaps = 20, alpha = 0.7)
```
Q11: Do the results from SMR and Susie agree? Explain!
the results do not seem to agree, i got totally different results form the different methods.

I can not identify any "causal" SNP:s that agree between the SMR and SUSIE results. THis is not completley unexpected since the methods use different traits, SUSIE looks at fine mapping with LD structure, and SMR focuses on strong eQTL associations that also pass the HEIDI test. This difference may be due to that tere are several causal variants in this region or because none of them are actually causal, which demonstrates that the methods can never be certain. it could be the case that all of the identified variants are causal, only some or perhhaps none.


Variant1
rs1319945
There are strong arguments for this variant mainly due to its strong results in the SuSiE analysis. It is in a single variant credible set with a PIP of 0.99767813 which means that according to the LD structure it is highly likley to be a causal variant for human height. It is said to be located in an intron which could be indicative of gene expression regulation according to the VEP search, however no studies on pubmed was found to be related to the variant. The strongest argument against this candidate is that it was not found in the SMR analysis.

Variant2
rs1873191
identified in study for humen height. Part of a credible set by susie, of which the combination was aroun 99% probable. not found in the SMR.

Variant 3
rs549201. 
Identified as as strong candidate by the smr analysis with a p_heidi > 0.05 and very strong p_SMR. not identified in susie analyis. Identified in human height study by VEP search.
